docs/adr/adr-005-data-retention.md

```diff
@@ -0,0 +1,186 @@
+# Decision Record: Data Retention
+
+---
+
+## Context
+
+The original ADR concerning data retention resides [here][initial policy] and
+outlines our approach from February 2018. A lot has changed since then, namely
+how we handle unused data that no longer serves a purpose in the system.
+
+We originally retained all data produced by EmailAlertApi (apart from emails
+which are [archived] to S3 & [deleted] after a week). This included subscribers
+which have been [nullified], ended subsciptions and subscriber lists with no
+subscriptions to name a few.
+
+These records were originally kept forever as we didn't know what data was
+useful. Now, nearly 3 years on, we have a pretty good understanding of what data
+we should keep and what we should remove from the system completely.
+
+Our primary motivation is to increase the amount of digest subscriptions we
+have versus individual subscriptions. This prompted a general review of data
+retention as we knew that the system already produced a sizeable amount of
+digest run subscribers (400k a day) which don't get cleaned up and just reside
+in the db.
+
+The benefit of removing unused data which is deemed as historic is that we can
+keep the db tidy and relevant. Perpetually keeping this data results in a
+detrimental effect on the db performance and complicates our auditing and
+analysis. An example of how dead data can skew our analytics can be found when
+we look at subscriber lists. We have over ~73,000 of these lists with active
+subscribers, but in the last week alone, we've accumulated 1139 lists which
+don't have any subscriptions at all (ended or active). These dead lists prevent
+us from doing simple things easily, like determining which lists are actually
+in use.
+
+[archived]: ...
+[deleted]: ...
+[nullified]: ...
+[initial policy]: https://github.com/alphagov/email-alert-api/blob/master/docs/adr/adr-003-initial-data-retention.md
+
+---
+
+## Decision
+
+The decision we came to was to implement a [worker] which runs periodically
+every day at [midday] to remove data we classify as historic. Removing this
+data was deemed as non urgent and so deleting once a day, in hours, seemed
+suffice.
+
+To deem data as historic we had to be ensure that the system no longer
+uses the data, and we don't need the data for future analytics.
+
+We decided:
+
+* On a max retention of 1 year for everything that's unused by the system.
+
+* On a shorter retention of 7 days for subscriber lists with no subscriptions,
```

Comment:
We should explain why here. (We want to get rid of unused lists ASAP to avoid them polluting our analytics.)

Reply:
Have added an explanation, pointing back to the context where I've given an example.